client/layout/logged-out.jsx

```diff
@@ -73,12 +81,15 @@ LayoutLoggedOut.displayName = 'LayoutLoggedOut';
 LayoutLoggedOut.propTypes = {
 	primary: PropTypes.element,
 	secondary: PropTypes.element,
+	// Connected props
+	hideMasterbar: PropTypes.bool,
 	section: PropTypes.oneOfType( [ PropTypes.bool, PropTypes.object ] ),
 	redirectUri: PropTypes.string,
 	showOAuth2Layout: PropTypes.bool,
 };
 
 export default connect( state => ( {
+	hideMasterbar: shouldHideMasterbar( state ),
```

Comment:
nitpick on naming: we often prefer boolean-ish names like `masterbarIsHidden` or even better, boolean-ish names _in the positive_ to prevent things like double-negative… `masterbarIsVisible`

To me, the second example below is clearer:

```js
! hideMasterbar, hideMasterbar

masterbarIsVisible, ! masterbarIsVisible
```

Reply:
https://s3-us-west-2.amazonaws.com/assets.coderag.prd/wp-content/uploads/2017/05/pizza.png

> convention of naming things so they will be used in the positive

naming is hard for sure. I'm guessing that the burden falls more on naming in the negative though just because we tend to use things more broadly than we expect when designing them. I think that's evident in this very PR where we were already double-negating the name.

hiding states are more complicated because `hidden` is positive. in this case it felt subjectively marginally worse to me than `visible` 🤷‍♂️